fuzz/fuzz_targets/fuzz_common.rs

```diff
@@ -95,6 +104,26 @@ where
         };
     }
 
+    let original_stdin_fd = if let Some(input_str) = pipe_input {
+        // we have pipe input
+        let mut input_file = tempfile::tempfile().unwrap();
+        write!(input_file, "{}", input_str).unwrap();
+        input_file.seek(SeekFrom::Start(0)).unwrap();
+
+        // Redirect stdin to read from the in-memory file
+        let original_stdin_fd = unsafe { dup(STDIN_FILENO) };
+        if original_stdin_fd == -1 || unsafe { dup2(input_file.as_raw_fd(), STDIN_FILENO) } == -1 {
+            return CommandResult {
+                stdout: "Failed to set up stdin redirection".to_string(),
+                stderr: "".to_string(),
```

Comment:
Is this on purpose that `stdout` contains the error message and `stderr` is empty?